### PR TITLE
MULE-13690 Adding a new expression language method to enable better l…

### PR DESCRIPTION
**Pull request by machaval**
…ogging

### CODE REVIEW

File: src/main/java/org/mule/runtime/api/el/ExpressionLanguage.java [hunk 1]
```diff
@@ -53,6 +53,20 @@
...
   /**
+   * Evaluates an expression according to a given {@link BindingContext}, the global one and the {@link DataType} of the expected
+   * result.
+   *
+   * The evaluation of this script will do a best effort to avoid failing when the result value can not be represented in the corresponding format.
+   *
+   * @param expression         the EL expression
+   * @param context            the current dynamic binding context to consider
+   * @return the result of the expression plus its type
+   * @throws ExpressionExecutionException or during transformation or during transformation
```
Comment (afelisatti) [inline src/main/java/org/mule/runtime/api/el/ExpressionLanguage.java]: throws seems incorrect

File: src/main/java/org/mule/runtime/api/el/ExpressionLanguage.java [hunk 2]
```diff
@@ -53,6 +53,20 @@
       throws ExpressionExecutionException;
 
   /**
+   * Evaluates an expression according to a given {@link BindingContext}, the global one and the {@link DataType} of the expected
```
Comment (afelisatti) [inline src/main/java/org/mule/runtime/api/el/ExpressionLanguage.java]: There's no DataType here.

File: src/main/java/org/mule/runtime/api/el/ExpressionLanguage.java (hunk 1 above)
Comment (machaval) [reply src/main/java/org/mule/runtime/api/el/ExpressionLanguage.java]: Why? An exception can still happen.

**Comment by aiannucci** [comment]
jenkins validate

**Pull request merged**